core/trino-main/src/main/java/io/trino/execution/scheduler/NodeScheduler.java

```diff
@@ -63,11 +66,14 @@ public NodeSelector createNodeSelector(Session session)
         return nodeSelectorFactory.createNodeSelector(requireNonNull(session, "session is null"));
     }
 
-    public static List<InternalNode> getAllNodes(NodeMap nodeMap, boolean includeCoordinator)
+    public static Set<InternalNode> getAllNodes(NodeMap nodeMap, boolean includeCoordinator)
```

Comment:
NodeScheduler changes look like they belong to a separate PR, its not related to the planner commits up to this point

Reply:
Dropped this commit